Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -1463,7 +1463,8 @@ static EditorEventListener()
         /// Used to register newly created instances in edit mode.
         /// Initially handled by using ExecuteAlways, but this attribute causes the instance to be destroyed as we enter play mode, which is disruptive to services.
         /// </summary>
-        private void OnValidate()
+        [UnityEditor.Callbacks.DidReloadScripts] // This is workaround for a known unity issue when calling refresh assetdatabase from inside a on validate scope.
```

Comment:
I wonder if we can get by with keeping `OnValidate` but using [`EditorApplication.delayCall`](https://docs.unity3d.com/ScriptReference/EditorApplication-delayCall.html) to postpone it outside the `OnValidate` scope?

Reply:
That works!